File: go/easy/1394_find_lucky_integer_in_an_array.go
func findLucky(arr []int) int {
    freq := make(map[int]int)

    for _, el := range arr {
        freq[el]++
    }

    res := -1
    for k, v := range freq {
        if k == v && k > res {
            res = k
        }
    }

    return res
}

File: rust/easy/1394_find_lucky_integer_in_an_array.rs
use std::collections::HashMap;

impl Solution {
    pub fn find_lucky(arr: Vec<i32>) -> i32 {
        let mut map = HashMap::new();
        for &el in &arr {
            *map.entry(el).or_insert(0) += 1;
        }
        map.into_iter()
            .filter(|(k, v)| k == v)
            .map(|(k, _)| k)
            .max()
            .unwrap_or(-1)
    }
}
